app/src/main/java/com/concordia/mcga/adapters/POISearchAdapter.java

```diff
@@ -0,0 +1,283 @@
+package com.concordia.mcga.adapters;
+
+import android.content.Context;
+import android.support.v7.widget.AppCompatImageView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseExpandableListAdapter;
+import android.widget.TextView;
+
+import com.concordia.mcga.activities.R;
+import com.concordia.mcga.models.Building;
+import com.concordia.mcga.models.Campus;
+
+import java.util.ArrayList;
+
+import de.hdodenhof.circleimageview.CircleImageView;
+
+/**
+ * This adapter searches through both campuses to find POIs (for now, buildings) that match a query.
+ */
+public class POISearchAdapter extends BaseExpandableListAdapter {
+    private Context context;
+
+    public static final int SGW_INDEX = 0;
+    public static final int LOYOLA_INDEX = 1;
+
+    private ArrayList<Building> sgwFilteredList;
+    private ArrayList<Building> loyolaFilteredList;
+
+    /**
+     * The constructor initializes the empty lists that store the currently queried POIs
+     * @param context View context to search
+     */
+    public POISearchAdapter(Context context) {
+        this.context = context;
+
+        this.sgwFilteredList = new ArrayList<>();
+        this.loyolaFilteredList = new ArrayList<>();
+    }
+
+    /**
+     * Evaluates the number of campuses active based on the current queried POIs
+     * @return The number of active campuses
+     */
+    @Override
+    public int getGroupCount() {
+        boolean sgwClear = getGroupIsEmpty(SGW_INDEX);
+        boolean loyClear = getGroupIsEmpty(LOYOLA_INDEX);
+
+        if (sgwClear && loyClear) {
+            return 0;
+        } else if (sgwClear || loyClear) {
+            return 1;
+        } else {
+            return 2;
+        }
+    }
+
+    /**
+     * This overrides the BaseExpandableList function to get the number of children (POI) entries
+     * @param i Campus index
+     * @return Number of POIs specific to the passed campus
+     */
+    @Override
+    public int getChildrenCount(int i) {
+        Campus camp = (Campus)getGroup(i);
+        if (camp == Campus.SGW) {
+            return sgwFilteredList.size();
+        } else if (camp == Campus.LOY) {
+            return loyolaFilteredList.size();
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * This overrides the BaseExpandableList function to get the campus specified by an index.
+     * This function is written specifically to address the quirks in the way BaseExpandableList
+     * handles groups and children - it is used internally to get the "group" (campus) based on the
+     * number of visible groups, so we have to handle different cases depending on whether the
+     * POIs filtered by the query come from a single campus or both of them.
+     * @param i Active campus index
+     * @return The campus specified by the index
+     */
```

Comment:
JavaDoc doesn't really explain what the method does, yes it returns an campus object, based on the campus index and based on if the campus group is empty, but how?

 Example:
> If both campus groups are empty, return null **Most probably this test for both empty groups isn't even needed because of the logic below, it would default to return null, anyways I'm following the logic from the beginning of the method**
> If SGW campus index was selected and it the SGW group is not empty return Campus.SGW object
> If LOY campus index was selected and it the LOY group is not empty return Campus.LOY object
> Else return null

I think this is what you tried to do, but there are too many conditionals at present to achieve this.